functions/src/winloss.ts

```diff
@@ -0,0 +1,26 @@
+import * as functions from "firebase-functions";
+import { lobbyCollection } from "./firestore-collections";
+import { isLobbyRequest } from "./firestore-functions-types";
+import { Lobby } from "./firestore-types/lobby";
+
+export const lobbyReturn = functions.https.onCall(async (data: unknown, context) => {
+  if (context.auth === undefined) {
+    return { error: "Not Signed In" };
+  }
+
+  // validate code
+  if (!isLobbyRequest(data)) {
+    return { error: "Invalid lobby code!" };
+  }
+  // get lobby doc
+  const lobby = await lobbyCollection.doc(data.code).get();
+  if (lobby.exists === false) {
+    return { error: "Lobby doesn't exist!" };
+  }
+
+  const { uids } = lobby.data() as Lobby;
+  if (context.auth.uid !== uids[0]) {
+    return { error: "Not allowed to return to lobby!" };
+  }
+  return lobbyCollection.doc(data.code).update({ state: "WAIT" });
```

Comment:
we should also some code to delete private players and set everyone to alive and maybe some other stuff I am forgeting